File: Day 29 : Bitwise AND/Solution.c
#include <stdio.h>

int bitwiseAnd(int N, int K) 
{
    int tempList[N];
    int i, j, temp, max = 0;
    
    for(i = 1; i <= N; i++)
    {
        tempList[i-1] = i;
    }
    
    for(i = 0; i < N; i++)
    {
        for(j = i + 1; j < N; j++)
        {
            temp = tempList[i] & tempList[j];
            if(temp < K && temp > max)
            {
                max = temp;
            }
        }
    }
    return max;
}

int main()
{
    int i, t;
    scanf("%d", &t);
  
    for (i = 0; i < t; i++) 
    {
        int count, lim;
        scanf("%d %d", &count, &lim);
        int res = bitwiseAnd(count, lim);
        printf("%d\n", res);
    }
    return 0;
}

File: Day 29 : Bitwise AND/Solution.py
def bitwiseAnd(N, K):
    if (K-1 | K) <= N :
        return K-1
    else:
        return K-2

if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        first_multiple_input = input().split()
        count = int(first_multiple_input[0])
        lim = int(first_multiple_input[1])
        res = bitwiseAnd(count, lim)
        print(res)
